model/generate.go

```diff
@@ -275,17 +276,57 @@ func (g *GeneratedProject) saveNewBuildsAndTasks(ctx context.Context, v *Version
 	return nil
 }
 
-func (g *GeneratedProject) findVariantsToActivate() []string {
-	var toActivate []string
+type batchTimeTasksAndVariants struct {
+	tasks    map[string][]string // tasks by variant that have batchtime
+	variants []string            // variants that have batchtime
+}
+
+func newBatchTimeTasksAndVariants() batchTimeTasksAndVariants {
+	return batchTimeTasksAndVariants{
+		tasks:    map[string][]string{},
+		variants: []string{},
+	}
+}
+
+func (b *batchTimeTasksAndVariants) isBatchTimeVariant(variant string) bool {
+	return utility.StringSliceContains(b.variants, variant)
+}
+
+func (b *batchTimeTasksAndVariants) batchTimeTasks(variant string) []string {
+	return b.tasks[variant]
+}
+
+func (b *batchTimeTasksAndVariants) hasAnyBatchTimeTasks() bool {
+	for _, tasks := range b.tasks {
+		if len(tasks) > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+// given some list of tasks, returns the tasks that don't have batchtime
+func (b *batchTimeTasksAndVariants) tasksWithoutBatchTime(taskNames []string, variant string) []string {
+	tasksWithoutBatchTime, _ := utility.StringSliceSymmetricDifference(taskNames, b.tasks[variant])
```

Comment:
it makes me really happy to see this function being used outside of the weird thing I created it for 😄